### PR TITLE
Fix CountBitLenForFloat on arm64

### LINKED ISSUE #2497: [Bug]: `TestCountBitLenForFloat` failed on darwin with arm64
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 2362d3149588023718be14d6df78f09ba5eedcde
- Hardware parameters: M1 chip
- OS type: macOS v12.0.1
- Others:
```

### Actual Behavior

`TestCountBitLenForFloat` failed on darwin with arm64:
```bash
go test -run=TestCountBitLenForFloat -race github.com/matrixorigin/matrixone/pkg/vectorize/bin
--- FAIL: TestCountBitLenForFloat (0.00s)
    bin_test.go:50:
                Error Trace:    bin_test.go:50
                Error:          Not equal:
                                expected: 135
                                actual  : 9
                Test:           TestCountBitLenForFloat
FAIL
FAIL    github.com/matrixorigin/matrixone/pkg/vectorize/bin     0.467s
FAIL
```

### Expected Behavior

`TestCountBitLenForFloat` should be ok on arm64.

### Steps to Reproduce

```bash
# On macOS with arm64
go test -run=TestCountBitLenForFloat -race github.com/matrixorigin/matrixone/pkg/vectorize/bin
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by e11jah**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Closes #2497

**What this PR does / why we need it:**

fix mistakes of built-in function bin() on arm64, see details in cr of matrixorigin/matrixone#2513

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by e11jah** [comment]
@broccoliSpicy PTAL

File: pkg/vectorize/bin/bin.go [hunk 1]
```diff
@@ -15,59 +15,109 @@
...
-		// convert to uint64 to count bits
-		val := uint64(x)
-		if val == 0 {
+		if x == 0 {
+			// special case for zero
+			r += 1
+		} else if x > 0 {
+			r += int64(bits.Len64(uint64(x)))
+		} else {
+			// calc two's complement for negative number to improve the compatibility between arm64 and x86 platforms
```
Comment (broccoliSpicy) [inline pkg/vectorize/bin/bin.go:64]: for negative integers, isn't the result length is fixed at 64? perhaps there are some redundant code here

File: pkg/vectorize/bin/bin.go [hunk 2]
```diff
@@ -15,59 +15,109 @@
...
+			mask := big.NewInt(1)
+			mask.Lsh(mask, 64)
+			n.Add(n, mask)
+
+			r += int64(n.BitLen())
+		}
+	}
+	return r
+}
+
```
Comment (broccoliSpicy) [inline pkg/vectorize/bin/bin.go:75]: seems like calculating the result length is quite a lot of cpu time, what if we don't calculate the result length ahead and use the slice's append function instead

**Comment by broccoliSpicy** [comment]
you know what, I am quite frustrated that we have to follow all the nonsenses mysql has, what does bin(6.2831) even mean?
 if you want, just drop the support for float types, your call @e11jah

**Comment by e11jah** [comment]
> you know what, I am quite frustrated that we have to follow all the nonsenses mysql has, what does bin(6.2831) even mean? if you want, just drop the support for float types, your call @e11jah

Indeed. The behavior of the bin function given the float numbers is weird, I don't know if there's anyone who will use it with the float types. Maybe the support for float types should be dropped.

**Comment by e11jah** [comment]
> you know what, I am quite frustrated that we have to follow all the nonsenses mysql has, what does bin(6.2831) even mean? if you want, just drop the support for float types, your call @e11jah

The support for float types has been removed.

**Comment by fengttt** [comment]
Close to free resource on ci.

**Pull request closed without merging**